### PR TITLE
ARM64-SVE: LeadingSignCount, LeadingZeroCount, PopCount

### PR DESCRIPTION
**Pull request by a74nh**
_SveMasklessUnaryOpTestTemplate is the same as _SveUnaryOpTestTemplate, but without the Conditional Select tests.

For all these, the API method has a different Op1 and Return type and so does not fit into a standard conditional select test.

For LeadingSignCount, LeadingZeroCount the underlying instruction does not have a mask, so can't be optimised using conditional select.

### CODE REVIEW

**Comment by a74nh** [comment]
I get some failures in the stress test due to `isLowPredicateRegister()` asserts, even though the hwintrinsiclist is correct. I'm assuming this will be fixed by https://github.com/dotnet/runtime/pull/102543

In the meantime, marking this as ready for review.

@dotnet/arm64-contrib @kunalspathak

**Review by kunalspathak — APPROVED** [review]
LGTM. Thanks!

File: src/coreclr/jit/hwintrinsiclistarm64sve.h [hunk 1]
```diff
@@ -68,6 +68,8 @@ HARDWARE_INTRINSIC(Sve,           FusedMultiplyAddNegated,
...
 HARDWARE_INTRINSIC(Sve,           FusedMultiplySubtractNegated,                                     -1,     -1,      false, {INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_sve_fnmls,      INS_sve_fnmls},   HW_Category_SIMD,                  HW_Flag_Scalable|HW_Flag_EmbeddedMaskedOperation|HW_Flag_HasRMWSemantics|HW_Flag_LowMaskedOperation|HW_Flag_FmaIntrinsic|HW_Flag_SpecialCodeGen)
+HARDWARE_INTRINSIC(Sve,           LeadingSignCount,                                                 -1,     -1,      false, {INS_sve_cls,        INS_invalid,        INS_sve_cls,        INS_invalid,        INS_sve_cls,        INS_invalid,        INS_sve_cls,        INS_invalid,        INS_invalid,        INS_invalid},     HW_Category_SIMD,                  HW_Flag_Scalable|HW_Flag_BaseTypeFromFirstArg|HW_Flag_EmbeddedMaskedOperation)
```
Comment (kunalspathak) [inline src/coreclr/jit/hwintrinsiclistarm64sve.h:71]: these 2 are missing the flag for `HW_Flag_LowMaskedOperation`

Comment (kunalspathak) [reply src/coreclr/jit/hwintrinsiclistarm64sve.h:71]: [CLS](https://docsmirror.github.io/A64/2023-06/cls_z_p_z.html) and [CLZ](https://docsmirror.github.io/A64/2023-06/clz_z_p_z.html)

**Comment by kunalspathak** [comment]
/ba-g Issues seems to be https://github.com/dotnet/runtime/issues/100174

File: src/coreclr/jit/hwintrinsiclistarm64sve.h (hunk 1 above)
Comment (a74nh) [reply src/coreclr/jit/hwintrinsiclistarm64sve.h]: I really did think I had added that! Thanks for fixing.

**Pull request merged**